runtime/error.c

```diff
@@ -116,13 +128,26 @@ static int storePC(void *data, PC pc) {
     return SUCCESS;
 }
 
-void _bal_error_backtrace_print(ErrorPtr ep) {
+void _bal_error_backtrace_print(ErrorPtr ep, uint32_t start) {
     GC PC *pcs = ep->pcs;
     uint32_t nPCs = ep->nPCs;
     BacktraceError err = { ep->backtraceErrorCode, ep->backtraceErrorMessage };
-    for (uint32_t i = 0; i < nPCs; i++) {
-        backtrace_pcinfo(state, pcs[i], printBacktraceLine, storeBacktraceError, &err);
+    if (start < nPCs) {
+        BacktraceStartLine backtraceStartLine = { err, ep->lineNumber };
+        backtrace_pcinfo(state, pcs[start], printBacktraceStartLine, storeBacktraceError, &backtraceStartLine);
+        err.code = backtraceStartLine.error.code;
+        err.message = backtraceStartLine.error.message;
+        if (err.code == NO_DEBUG_INFO) {
+            backtrace_syminfo(state, pcs[start], printMissingSymbols, storeBacktraceError, &err);
+        }
+        for (uint32_t i = start + 1; i < nPCs; i++) {
+            backtrace_pcinfo(state, pcs[i], printBacktraceLine, storeBacktraceError, &err);
```

Comment:
If you got NO_DEBUG_INFO error above, do you want to keep calling pcinfo?

Reply:
It is not needed, changed.

Reply:
Are you sure? What happens if in a frame that contains functions from multiple modules, some modules have debug info and some don't?

Reply:
```
   As a special case, the ERRNUM argument will be passed as -1 if no
   debug info can be found for the executable, or if the debug info
   exists but has an unsupported version, but the function requires
   debug info (e.g., backtrace_full, backtrace_pcinfo).  The MSG in
   this case will be something along the lines of "no debug info".
   Similarly, ERRNUM will be passed as -1 if there is no symbol table,
   but the function requires a symbol table (e.g., backtrace_syminfo).
   This may be used as a signal that some other approach should be
   tried. 
```


Based on this libbacktrace comment, what I understood was, errnum -1 comes if the whole executable does not have debug info or symbol table. In the same source file, if some functions do not have debug info, libbacktrace does not call error_callback. Let me check what happens when we have multiple modules.